src/ws/channel.rs

```diff
@@ -1,4 +1,6 @@
-#[derive(Eq, PartialEq, Clone, Debug)]
+use thiserror::Error as ThisError;
+
+#[derive(Eq, PartialEq, Clone, Debug, ThisError)]
```

Comment:
We should not be deriving `thiserror::Error` for `Channel`, because `Channel` is not an error type.

Reply:
fixed